src/main/java/uk/gov/hmcts/ccd/domain/model/search/elasticsearch/ElasticsearchRequest.java

```diff
@@ -4,39 +4,73 @@
 import com.fasterxml.jackson.databind.node.ArrayNode;
 import com.fasterxml.jackson.databind.node.BooleanNode;
 import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
 import lombok.Data;
 import lombok.NonNull;
 import uk.gov.hmcts.ccd.data.casedetails.search.MetaData;
+import uk.gov.hmcts.ccd.endpoint.exceptions.BadSearchRequest;
 
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
 import java.util.Objects;
-import java.util.stream.Collectors;
 import java.util.stream.StreamSupport;
 
+import static java.util.stream.Collectors.toList;
+import static uk.gov.hmcts.ccd.config.JacksonUtils.MAPPER;
+import static uk.gov.hmcts.ccd.data.casedetails.CaseDetailsEntity.DATA_CLASSIFICATION_COL;
+import static uk.gov.hmcts.ccd.data.casedetails.CaseDetailsEntity.DATA_COL;
+
 @Data
 public class ElasticsearchRequest {
 
     public static final String SORT = "sort";
     public static final String SOURCE = "_source";
     public static final String QUERY = "query";
+    public static final String NATIVE_ES_QUERY = "native_es_query";
+    public static final String SUPPLEMENTARY_DATA = "supplementary_data";
     public static final String CASE_DATA_PREFIX = "data.";
+    public static final String SUPPLEMENTARY_DATA_PREFIX = "supplementary_data.";
     public static final String COLLECTION_VALUE_SUFFIX = ".value";
     public static final String WILDCARD = "*";
+    public static final ArrayNode METADATA_FIELDS;
+
+    private JsonNode nativeSearchRequest;
+    private ArrayNode supplementaryData;
+
+    static {
+        METADATA_FIELDS = MAPPER.createArrayNode();
+        Arrays.stream(MetaData.CaseField.values())
+            .map(field -> new TextNode(field.getDbColumnName()))
```

Comment:
maybe a method Metada.getColumnNames()?

Reply:
furthermore, can't this be part of the constructor? how come a static block?

Reply:
It could have been part of the constructor, but chose not to since a new `ElasticsearchRequest` instance will be created every time a search is conducted but the metadata fields won't change.